packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+    };
+  }
+
+  get helperTextAdapter() {
+    return {
+      showToScreenReader: () =>
+        this.setState({helperTextShowToScreenReader: !this.state.helperTextShowToScreenReader}),
+      setValidity: (isValid) => this.setState({isValid}),
+    };
+  }
+
+  getIsRtl = () => {
+    if (this.textFieldElement.current) {
+      const dir = window.getComputedStyle(this.textFieldElement.current).getPropertyValue('direction');
+      return dir === 'rtl';
+    }
+  }
+
+  inputProps(props) {
+    return Object.assign({}, props, {
+      foundation: this.foundation_,
+      handleFocusChange: (isFocused) => this.setState({isFocused}),
+      handleValueChange: (value) => this.setState({value}),
+      // These are callbacks for Input, which set validity.badInput &
+      // validity.valid.
+      setBadInputHandler: (getBadInput) => this.getBadInput = getBadInput,
+      setIsValidHandler: (getIsValid) => this.getIsValid = getIsValid,
+      setDisabled: (disabled) => this.setState({disabled}),
+      setInputId: (id) => this.setState({inputId: id}),
+    });
+  }
+
+  /**
+  * render methods
+  */
+
+  render() {
+    const {
+      label,
+      fullWidth,
+      helperText,
+      outlined,
+      leadingIcon,
+      trailingIcon,
+      textarea,
+    } = this.props;
+
+    const textField = (
+      <div
+        className={this.classes}
+        onClick={() => this.foundation_ && this.foundation_.handleTextFieldInteraction()}
+        onKeyDown={() => this.foundation_ && this.foundation_.handleTextFieldInteraction()}
+        key='text-field-container'
+        ref={this.textFieldElement}
+      >
+        {leadingIcon ? this.renderIcon(leadingIcon) : null}
+        {this.renderInput()}
+        {label && !fullWidth ? this.renderLabel() : null}
+        {outlined ? this.renderNotchedOutline() : null}
+        {!fullWidth && !textarea && !outlined ? this.renderLineRipple() : null}
+        {trailingIcon ? this.renderIcon(trailingIcon) : null}
+      </div>
+    );
+
+    if (helperText) {
+      return ([
+        textField, this.renderHelperText(),
+      ]);
+    }
+    return textField;
+  }
+
+  renderInput() {
+    const child = React.Children.only(this.props.children);
+    const props = this.inputProps(child.props);
+    return React.cloneElement(child, props);
+  }
+
+  renderLabel() {
+    const {label, floatingLabelClassName} = this.props;
+    const {inputId} = this.state;
+    return (
+      <FloatingLabel
+        className={floatingLabelClassName}
+        float={this.state.shouldFloatLabel}
+        handleWidthChange={
+          (labelWidth) => this.setState({labelWidth})}
+        ref={this.floatingLabelElement}
+        htmlFor={inputId}
+      >
+        {label}
+      </FloatingLabel>
+    );
+  }
+
+  renderLineRipple() {
+    const {lineRippleClassName} = this.props;
+    const {activeLineRipple, lineRippleCenter} = this.state;
+    return (
+      <LineRipple
+        rippleCenter={lineRippleCenter}
```

Comment:
Okay.. maybe 'lineRippleCenter' is as clear as 'rippleCenter'...not sure what units this is in though